### PR TITLE
vomitgoose singularity

### PR DESCRIPTION
**Pull request by plapatin**
## About The Pull Request
vomitgoose singularity. eats things. spits vomit.
goose

## Why It's Good For The Game
vomitgoose singularity

## Changelog
:cl: plapatin, sprites by mrdoombringer
add: vomitgoose singularity
/:cl:

todo:

- [x] fix that annoying bug where the goose won't eat mobs
- [x] add a way to summon the gooseulo in-game

### CODE REVIEW

File: code/modules/power/singularity/singularity.dm [hunk 1]
```diff
@@ -446,3 +447,37 @@
...
+	grav_pull = 0 //How many tiles out do we pull?
+	consume_range = 5 //How many tiles out do we eat
+	CanSupermatter = 0
+	light_power = 0.7
+	light_range = 7
+	light_color = rgb(0, 255, 0)
+
+/obj/singularity/vomit/consume(atom/A)
+	var/turf/Tgoose = get_turf(src)
+	if (istype(A, /obj/effect/decal/cleanable/vomit))
```
Comment (Cyprex) [inline code/modules/power/singularity/singularity.dm]: Move this to the start of the proc, i. e. up a line.

**Comment by Arathianen** [comment]
If a SM delams and it eats a bunch of geese, and the delaming would produce a singulo, it should produce a vomitgoose singularity

Something like that.

**Comment by Quartich** [comment]
> If a SM delams and it eats a bunch of geese, and the delaming would produce a singulo, it should produce a vomitgoose singularity
> 
> Something like that.

@Arathianen  I just imagined hearing that out of context

File: code/modules/power/singularity/singularity.dm [hunk 2]
```diff
@@ -26,6 +26,7 @@
 	var/target = null //its target. moves towards the target if it has one
 	var/last_failed_movement = 0//Will not move in the same dir if it couldnt before, will help with the getting stuck on fields thing
 	var/last_warning
+	var/CanSupermatter = 1 //can it actually eat the supermatter
```
Comment (JJRcop) [inline code/modules/power/singularity/singularity.dm]: `TRUE`

File: code/modules/power/singularity/singularity.dm [hunk 3]
```diff
@@ -289,7 +290,7 @@
 /obj/singularity/proc/consume(atom/A)
 	var/gain = A.singularity_act(current_size, src)
 	src.energy += gain
-	if(istype(A, /obj/machinery/power/supermatter_crystal) && !consumedSupermatter)
+	if (CanSupermatter == 1 && istype(A, /obj/machinery/power/supermatter_crystal) && !consumedSupermatter)
```
Comment (JJRcop) [inline code/modules/power/singularity/singularity.dm]: `TRUE`

File: code/modules/power/singularity/singularity.dm [hunk 4]
```diff
@@ -446,3 +447,37 @@
...
+	pixel_x = -96
+	pixel_y = -96
+	density = FALSE
+	current_size = 9 //It moves/eats like a max-size singulo, aside from range.
+	contained = 0 //Are we going to move around?
+	dissipate = 0 //Do we lose energy over time?
+	move_self = 1 //Do we move on our own?
+	grav_pull = 0 //How many tiles out do we pull?
+	consume_range = 5 //How many tiles out do we eat
+	CanSupermatter = 0
```
Comment (JJRcop) [inline code/modules/power/singularity/singularity.dm]: `FALSE`

**Review by JJRcop — COMMENTED** [review]
If `CanSupermatter` is a bool, you should use the bool macros.

**Comment by OverlyGeneric** [comment]
honestly, it shouldn't be hard to make if a singularity is rampaging through the station.

if it eats vomitgoose then it turns into a goosingularity but if it eats an SM shard along with it...

File: code/modules/power/singularity/singularity.dm [hunk 5]
```diff
@@ -446,3 +447,37 @@
...
+	light_power = 0.7
+	light_range = 7
+	light_color = rgb(0, 255, 0)
+
+/obj/singularity/vomit/consume(atom/A)
+	if (istype(A, /obj/effect/decal/cleanable/vomit))
+		return
+	var/turf/Tgoose = get_turf(src)
+	var/turf/spawnloc = isturf(A) ? A : get_turf(A)
+	playsound(Tgoose, 'sound/effects/splat.ogg', 50, 1)
```
Comment (nemvar) [inline code/modules/power/singularity/singularity.dm]: ``playsound(src, ...`` and remove line 473.
Also the fourth argument is a boolean.

Comment (plapatin) [reply code/modules/power/singularity/singularity.dm]: i have to get the turf or else it won't play any sound

Comment (nemvar) [reply code/modules/power/singularity/singularity.dm]: wrong. https://github.com/tgstation/tgstation/blob/master/code/game/sound.dm

**Comment by wesoda25** [comment]
inb4 vomit goose breaks the game and is ignored for months

**Comment by IndieanaJones** [comment]
> by the way, if anybody here can figure out why this singularity won't eat mobs, please for the love of god tell me because i'm absolutely stumped on the issue

Someone might need to look into this, but I don't think singularities eat mobs right now on live.  I got sucked up and flung, but I was still alive.

**Comment by Detective-Google** [comment]
> > by the way, if anybody here can figure out why this singularity won't eat mobs, please for the love of god tell me because i'm absolutely stumped on the issue
> 
> Someone might need to look into this, but I don't think singularities eat mobs right now on live. I got sucked up and flung, but I was still alive.

Afaik the current singulo doesn't straight up eat people, it gibs them on contact and then usually eats the gibs

**Comment by dee-dubya** [comment]
The sprite still triggers me. The goose is vomiting backwards.

**Comment by plapatin** [comment]
this fix is about as hacky as it gets but i got it to eat mobs

**Comment by plapatin** [comment]
sm interaction soon

File: code/modules/power/singularity/singularity.dm [hunk 6]
```diff
@@ -446,3 +447,52 @@
...
+	..()
+
+/obj/singularity/vomit/attack_alien(mob/user)
+	..()
+
+/obj/singularity/vomit/attack_animal(mob/user)
+	..()
+
+/obj/singularity/vomit/attackby(obj/item/W, mob/user, params)
+	..()
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/power/singularity/singularity.dm]: nani?

Comment (ExcessiveUseOfCobblestone) [inline code/modules/power/singularity/singularity.dm]: ???

File: code/modules/power/singularity/singularity.dm [hunk 7]
```diff
@@ -289,7 +290,7 @@
 /obj/singularity/proc/consume(atom/A)
 	var/gain = A.singularity_act(current_size, src)
 	src.energy += gain
-	if(istype(A, /obj/machinery/power/supermatter_crystal) && !consumedSupermatter)
+	if (CanSupermatter == TRUE && istype(A, /obj/machinery/power/supermatter_crystal) && !consumedSupermatter)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/power/singularity/singularity.dm]:
```suggestion
	if (CanSupermatter && istype(A, /obj/machinery/power/supermatter_crystal) && !consumedSupermatter)
```

File: code/modules/power/singularity/singularity.dm (hunk 6 above)
Comment (plapatin) [reply code/modules/power/singularity/singularity.dm]: it won't eat mobs without these interactions (from my testing anyways)

Comment (ExcessiveUseOfCobblestone) [reply code/modules/power/singularity/singularity.dm]: it should be calling inher by default wtf

File: code/modules/power/supermatter/supermatter.dm [hunk 8]
```diff
@@ -291,10 +294,14 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
...
-			S.energy = 800
-			S.consume(src)
+	if(T)
+		var/obj/singularity/S = null
+		if(ate_goose == TRUE)
+			S = new /obj/singularity/vomit(T)
+		else
+			S = new /obj/singularity(T)
+		S.energy = 800
+		S.consume(src)
```
Comment (TheChosenEvilOne) [inline code/modules/power/supermatter/supermatter.dm]: Indentation, if I am looking at this correctly it turns into a singularity always even when normally exploding.

Comment (plapatin) [reply code/modules/power/supermatter/supermatter.dm]: didn't spawn any singularity when i tried a standard delamination

**Comment by PKPenguin321** [comment]
make it so the goose has a slight chance to puke up a beaten dead horse

**Comment by plapatin** [comment]
> 
> 
> make it so the goose has a slight chance to puke up a beaten dead horse

do you have a horse sprite? i'll do it

**Comment by ExcessiveUseOfCobblestone** [comment]
when you can figure out my review re-PR it

**Comment by Detective-Google** [comment]
F

**Comment by dee-dubya** [comment]
Let the goose be loose

**Pull request closed without merging**